### DISCUSSION TITLE
Issues loading java.lang objects into Soot with Scala

### DISCUSSION THREAD

**Discussion by eikNamo** [soot-oss/soot#1889] [Q&A]
__TL;DR__ - I'm having troubles making Soot resolve Java classpaths within Scala code. It seems that Soot cannot find java.lang.objects class when trying to load it without an indication where it is looking for it. 

The code that is being executed:
<pre>
import soot.{Modifier, Scene, SootClass}
import java.lang.Object

object SootPlayground extends App {
  Scene.v().loadClassAndSupport("java.lang.Object")
  val sClass = new SootClass("test", Modifier.PUBLIC)
  sClass.setSuperclass(Scene.v().getSootClass("java.lang.Object"))
  Scene.v().addClass(sClass)
}</pre>

The stack trace:
<pre>java.nio.file.NoSuchFileException: /modules/modules
	at java.base/jdk.internal.jrtfs.JrtFileSystem.checkNode(JrtFileSystem.java:486)
	at java.base/jdk.internal.jrtfs.JrtFileSystem.isDirectory(JrtFileSystem.java:431)
	at java.base/jdk.internal.jrtfs.JrtDirectoryStream.<init>(JrtDirectoryStream.java:58)
	at java.base/jdk.internal.jrtfs.JrtPath.newDirectoryStream(JrtPath.java:642)
	at java.base/jdk.internal.jrtfs.JrtFileSystemProvider.newDirectoryStream(JrtFileSystemProvider.java:311)
	at java.base/java.nio.file.Files.newDirectoryStream(Files.java:482)
	at soot.asm.AsmJava9ClassProvider.find(AsmJava9ClassProvider.java:50)
	at soot.SourceLocator.getClassSource(SourceLocator.java:187)
	at soot.SootResolver.bringToHierarchyUnchecked(SootResolver.java:231)
	at soot.SootResolver.bringToHierarchy(SootResolver.java:221)
	at soot.SootResolver.bringToSignatures(SootResolver.java:292)
	at soot.SootResolver.processResolveWorklist(SootResolver.java:179)
	at soot.SootResolver.resolveClass(SootResolver.java:141)
	at soot.Scene.loadClass(Scene.java:1009)
	...
</pre>

If I try to omit the __Scene.v().loadClassAndSupport("java.lang.Object")__ line, then the Soot clearly errors about missing classpath for __java.lang.object__. See bellow:
<pre>Exception in thread "main" java.lang.RuntimeException: 
Aborting: can't find classfile java.lang.Object
	at soot.Scene.getSootClass(Scene.java:1190)
	at initial_test.SootPlayground$.delayedEndpoint$initial_test$SootPlayground$1(SootPlayground.scala:11)
	at initial_test.SootPlayground$delayedInit$body.apply(SootPlayground.scala:6)
	at scala.Function0.apply$mcV$sp(Function0.scala:39)
	at scala.Function0.apply$mcV$sp$(Function0.scala:39)
	at scala.runtime.AbstractFunction0.apply$mcV$sp(AbstractFunction0.scala:17)
	at scala.App.$anonfun$main$1$adapted(App.scala:80)
	at scala.collection.immutable.List.foreach(List.scala:431)
	at scala.App.main(App.scala:80)
	at scala.App.main$(App.scala:78)
	at initial_test.SootPlayground$.main(SootPlayground.scala:6)
	at initial_test.SootPlayground.main(SootPlayground.scala)

Process finished with exit code 1
</pre>

Extra info:
Soot version: <pre>trunk</pre>

Command line (executed by IntelliJ IDEA):
<pre>C:\Users\mzdan\.jdks\corretto-17.0.3\bin\java.exe "-javaagent:C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2021.3.3\lib\idea_rt.jar=62981:C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2021.3.3\bin" -Dfile.encoding=UTF-8 -classpath C:\Users\mzdan\IdeaProjects\akka-quickstart-scala\thesis-tests\target\scala-2.12\classes;C:\Users\mzdan\AppData\Local\Coursier\cache\v1\https\repo1.maven.org\maven2\com\typesafe\akka\akka-actor_2.12\2.5.13\akka-actor_2.12-2.5.13.jar;C:\Users\mzdan\AppData\Local\Coursier\cache\v1\https\repo1.maven.org\maven2\com\typesafe\akka\akka-testkit_2.12\2.5.13\akka-testkit_2.12-2.5.13.jar;C:\Users\mzdan\AppData\Local\Coursier\cache\v1\https\repo1.maven.org\maven2\com\typesafe\config\1.3.2\config-1.3.2.jar;C:\Users\mzdan\AppData\Local\Coursier\cache\v1\https\repo1.maven.org\maven2\org\scala-lang\modules\scala-java8-compat_2.12\0.8.0\scala-java8-compat_2.12-0.8.0.jar;C:\Users\mzdan\AppData\Local\Coursier\cache\v1\https\repo1.maven.org\maven2\org\scala-lang\modules\scala-xml_2.12\2.0.1\scala-xml_2.12-2.0.1.jar;C:\Users\mzdan\.sbt\boot\scala-2.12.15\lib\scala-library.jar;C:\Users\mzdan\.sbt\boot\scala-2.12.15\lib\scala-reflect.jar;C:\Users\mzdan\AppData\Local\Coursier\cache\v1\https\repo1.maven.org\maven2\org\scalactic\scalactic_2.12\3.2.11\scalactic_2.12-3.2.11.jar;C:\Users\mzdan\AppData\Local\Coursier\cache\v1\https\repo1.maven.org\maven2\org\scalatest\scalatest-compatible\3.2.11\scalatest-compatible-3.2.11.jar;C:\Users\mzdan\AppData\Local\Coursier\cache\v1\https\repo1.maven.org\maven2\org\scalatest\scalatest-core_2.12\3.2.11\scalatest-core_2.12-3.2.11.jar;C:\Users\mzdan\AppData\Local\Coursier\cache\v1\https\repo1.maven.org\maven2\org\scalatest\scalatest-diagrams_2.12\3.2.11\scalatest-diagrams_2.12-3.2.11.jar;C:\Users\mzdan\AppData\Local\Coursier\cache\v1\https\repo1.maven.org\maven2\org\scalatest\scalatest-featurespec_2.12\3.2.11\scalatest-featurespec_2.12-3.2.11.jar;C:\Users\mzdan\AppData\Local\Coursier\cache\v1\https\repo1.maven.org\maven2\org\scalatest\scalatest-flatspec_2.12\3.2.11\scalatest-flatspec_2.12-3.2.11.jar;C:\Users\mzdan\AppData\Local\Coursier\cache\v1\https\repo1.maven.org\maven2\org\scalatest\scalatest-freespec_2.12\3.2.11\scalatest-freespec_2.12-3.2.11.jar;C:\Users\mzdan\AppData\Local\Coursier\cache\v1\https\repo1.maven.org\maven2\org\scalatest\scalatest-funspec_2.12\3.2.11\scalatest-funspec_2.12-3.2.11.jar;C:\Users\mzdan\AppData\Local\Coursier\cache\v1\https\repo1.maven.org\maven2\org\scalatest\scalatest-funsuite_2.12\3.2.11\scalatest-funsuite_2.12-3.2.11.jar;C:\Users\mzdan\AppData\Local\Coursier\cache\v1\https\repo1.maven.org\maven2\org\scalatest\scalatest-matchers-core_2.12\3.2.11\scalatest-matchers-core_2.12-3.2.11.jar;C:\Users\mzdan\AppData\Local\Coursier\cache\v1\https\repo1.maven.org\maven2\org\scalatest\scalatest-mustmatchers_2.12\3.2.11\scalatest-mustmatchers_2.12-3.2.11.jar;C:\Users\mzdan\AppData\Local\Coursier\cache\v1\https\repo1.maven.org\maven2\org\scalatest\scalatest-propspec_2.12\3.2.11\scalatest-propspec_2.12-3.2.11.jar;C:\Users\mzdan\AppData\Local\Coursier\cache\v1\https\repo1.maven.org\maven2\org\scalatest\scalatest-refspec_2.12\3.2.11\scalatest-refspec_2.12-3.2.11.jar;C:\Users\mzdan\AppData\Local\Coursier\cache\v1\https\repo1.maven.org\maven2\org\scalatest\scalatest-shouldmatchers_2.12\3.2.11\scalatest-shouldmatchers_2.12-3.2.11.jar;C:\Users\mzdan\AppData\Local\Coursier\cache\v1\https\repo1.maven.org\maven2\org\scalatest\scalatest-wordspec_2.12\3.2.11\scalatest-wordspec_2.12-3.2.11.jar;C:\Users\mzdan\AppData\Local\Coursier\cache\v1\https\repo1.maven.org\maven2\org\scalatest\scalatest_2.12\3.2.11\scalatest_2.12-3.2.11.jar;C:\Users\mzdan\IdeaProjects\akka-quickstart-scala\thesis-tests\lib\idea_rt.jar;C:\Users\mzdan\IdeaProjects\akka-quickstart-scala\thesis-tests\lib\jrt-fs.jar;C:\Users\mzdan\IdeaProjects\akka-quickstart-scala\thesis-tests\lib\sootclasses-trunk-jar-with-dependencies.jar initial_test.SootPlayground</pre>

Max Memory:
<pre>3623MB</pre>

**Comment by skyleaworlder** [comment]
Have you loaded `rt.jar` for Soot framework?

You can add the following lines of code before calling `Scene.v().loadClassAndSupport("java.lang.Object")`

```scala
val libs: String = "<your-path-to-rt-jar>" + File.separator + "rt.jar"
Options.v().set_soot_classpath(libs)
```
